test/test_datasets.py

```diff
@@ -9,6 +9,14 @@
 from fakedata_generation import mnist_root, cifar_root, imagenet_root
 
 
+def generic_dataset_test(tester, dataset, num_images=1, cls='fakedata'):
```

Comment:
Can you maybe rename this function to `basic_classification_dataset_test` or something that includes `classification` in it? Not all datasets have the `class_to_idx` attribute in it, so I'd not want (at least for now) to force people to add it just to make it compliant with this test. Maybe in the future we could have a `ClassificationDataset` that has this attribute, and then we could start enforcing the presence of this attribute.

Also, any particular reason why this is not a method in `Tester`? If it doesn't start with `test`, it won't be run by the `Tester`, so we can have helper functions inside `Tester`.

Reply:
> [...] many of our datasets do not return `(Image, int)`, so I'd rather not give the impression that they should.

Shame on me, I keep forgetting that.